### PR TITLE
set semantic-idle-summary-mode on hook

### PR DESCRIPTION
**Pull request by nbecker**
Thank you for contributing to Spacemacs!

Before you submit this pull request, please ensure it is against the develop branch and not master.

This message should be replaced with a description of your change.

Thank you <3

### CODE REVIEW

File: layers/+lang/python/funcs.el [hunk 1]
```diff
@@ -63,4 +63,6 @@
   " Disable semantic-idle-summary in Python mode.
 Anaconda provides more useful information but can not do it properly
 when this mode is enabled since the minibuffer is cleared all the time."
-  (semantic-idle-summary-mode 0))
+  (add-hook 'semantic-mode-hook (lambda()
+                                  (semantic-idle-summary-mode 0))
+            ))
```
Comment (NJBS) [inline layers/+lang/python/funcs.el]: Closing parentheses should not got on their own line.

https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html

**Comment by StreakyCobra** [comment]
A little bit of context or an explanation about the motivations of this change would be better than the default template message.

**Comment by nbecker** [comment]
Without this patch, attempt to enter python-mode resulted in an error: semantic-idle-summary-mode is undefined.

I don't know if this is actually the best fix.

**Comment by nixmaniack** [comment]
Disabling `semantic-idle-summary-mode` should be in `post-init-semantic` as per #5592. Looks like later [changes](https://github.com/syl20bnr/spacemacs/commit/6c2665e0f78ae82891b7e2b1ddacea70e9d81933) moved it to `anaconda-mode` `:init` block. So in case `semantic` layer isn't used, it'll throw an error. 

So I think it should be moved back to post-init-semantic with anaconda-mode guard as proposed in #5592. 

cc: @cpaulik

**Comment by cpaulik** [comment]
@nixmaniack Thanks, Yes I think it should be in `post-init-semantic`. @nbecker Your fix disables `idle-summary-mode` for all layers when the python layer is used. So if you want to work on it then please change your PR.

I did not really realize that @syl20bnr moved it although he said so in his comment. Also how is this merged? I don't see this commit in `develop`

**Comment by syl20bnr** [comment]
This is already merged but not pushed, I indeed moved some stuff to semantic post-init. I will push soon.

**Comment by syl20bnr** [comment]
Thank you ! 
Corrections were made in commit https://github.com/syl20bnr/spacemacs/commit/47776e8
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**